firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -118,17 +118,17 @@ public InstallationResponse createFirebaseInstallation(
         }
 
         int httpResponseCode = httpsURLConnection.getResponseCode();
-        switch (httpResponseCode) {
-          case 200:
-            return readCreateResponse(httpsURLConnection);
-          case 500:
-            retryCount++;
-            break;
-          default:
-            throw new FirebaseException(readErrorResponse(httpsURLConnection));
+
+        if (httpResponseCode == 200) {
+          return readCreateResponse(httpsURLConnection);
+        } else if (httpResponseCode >= 500 && httpResponseCode < 600) {
+          retryCount++;
+        } else {
+          throw new FirebaseException(readErrorResponse(httpsURLConnection));
         }
```

Comment:
I recommend rewriting this code like this to add a bit more code clarity:

```
if (httpResponseCode == 200) {
  return readCreateResponse(httpsURLConnection);
}
// usually the FIS server recovers from errors: We retry one time before giving up.
if (httpResponseCode >= 500 && httpResponseCode < 600) {
  retryCount++;
  continue;
}

// unrecoverable server response or unknown error
throw new FirebaseException(readErrorResponse(httpsURLConnection));

Reply:
Done. Thanks.